service/scaladsl/server/src/main/scala/com/lightbend/lagom/internal/scaladsl/server/ScaladslServiceRouter.scala

```diff
@@ -77,16 +78,24 @@ class ScaladslServiceRouter(override protected val descriptor: Descriptor, servi
     }
   }
 
+  private val config = ConfigFactory.load
+  private val default_log_level = config.getString("lagom.logging.default")
+  private val log_levels: Map[String, String] = config.getConfig("lagom.logging.exception.logging.level").root.unwrapped.asScala.toMap.mapValues(_.asInstanceOf[String])
+
   override protected def maybeLogException(exc: Throwable, log: => Logger, call: Call[_, _]) = {
-    exc match {
-      case _: NotFound | _: Forbidden => // no logging
-      case e @ (_: UnsupportedMediaType | _: PayloadTooLarge | _: NotAcceptable) =>
-        log.warn(e.getMessage)
-      case e =>
-        log.error(s"Exception in ${call.callId}", e)
+    translateErrorIntoLogLevel(exc) match {
+      case "TRACE" => log.trace(exc.getMessage)
+      case "DEBUG" => log.debug(exc.getMessage)
+      case "INFO"  => log.info(exc.getMessage)
+      case "WARN"  => log.warn(exc.getMessage)
+      case "ERROR" => log.error(s"Exception in ${call.callId}", exc)
     }
   }
 
+  def translateErrorIntoLogLevel(exc: Throwable): String = {
+    log_levels.getOrElse(exc.getClass.getName, default_log_level)
+  }
+
```

Comment:
I find it weird that this is public and then there's a test that needs all the complexity of building a `ScaladslServiceRouter` in order to test this. We could extract it into a separate `object` and use if from `ScaladslServiceRouter`.